app/models/grda_warehouse/tasks/scrub_pii/scrub_client_pii_task.rb

```diff
@@ -0,0 +1,115 @@
+
+        @scrubber.perform(client_scope)
+        client_scope.find_in_batches do |clients|
+          process_client_batch(clients)
+        end
+      end
+    end
+
+    protected
+
+    def process_client_batch(clients)
+      client_ids = clients.map(&:id)
+
+      # delete directly-related client records
+      delete_custom_data_elements(Hmis::Hud::Client, client_ids)
+      delete_versions([GrdaWarehouse::Hud::Client, Hmis::Hud::Client], client_ids)
+
+      clients.group_by(&:data_source_id).map do |data_source_id, ds_clients|
+        personal_ids = ds_clients.map(&:PersonalID)
+        delete_hmis_client_records(personal_ids, data_source_id)
+        delete_enrollment_related_records(personal_ids, data_source_id)
+      end
+    end
+
+    def delete_custom_data_elements(model, ids)
+      arel = Hmis::Hud::CustomDataElementDefinition.arel_table
+      conditions = [
+        arel[:label].matches('%SSN%', nil, true), # case sensitive
+        arel[:label].matches('%social security number%'),
+        arel[:label].matches('%DOB%', nil, true), # case sensitive
+        arel[:label].matches('%date of birth%'),
+        arel[:label].matches('%first name%'),
+        arel[:label].matches('%last name%'),
+        arel[:label].matches('%address%'),
+        arel[:label].matches('%city%'),
+        arel[:label].matches('%email%'),
+        arel[:label].matches('%phone%'),
+        arel[:label].matches('%license number%'),
+        arel[:label].matches('%policy number%'),
+      ].reduce(&:or)
+
+      scope = Hmis::Hud::CustomDataElement.with_deleted.
+        where(owner_type: model.sti_name, owner_id: ids).
+        joins(:data_element_definition).
+        merge(Hmis::Hud::CustomDataElementDefinition.with_deleted.where(conditions))
+
+      delete_versions([Hmis::Hud::CustomDataElement], scope.pluck(:id))
+      scope.delete_all
+    end
+
+    def delete_enrollment_related_records(personal_ids, data_source_id)
+      enrollments = GrdaWarehouse::Hud::Enrollment.
+        with_deleted.
+        where(data_source: data_source_id, PersonalID: personal_ids)
+      return if enrollments.to_a.empty?
+
+      enrollment_ids = enrollments.map(&:id)
+      delete_custom_data_elements(Hmis::Hud::Enrollment, enrollment_ids)
+    end
+
+    def delete_hmis_client_records(personal_ids, data_source_id)
+      # delete from custom tables that may contain sensitive info
+      [
+        Hmis::Hud::CustomClientAddress,
+        Hmis::Hud::CustomClientName,
+        Hmis::Hud::CustomClientContactPoint,
+        Hmis::Hud::CustomCaseNote,
+      ].each do |model|
+        scope = model.with_deleted.where(data_source_id: data_source_id, PersonalID: personal_ids)
+        delete_versions([model], scope.map(&:id))
+        scope.delete_all
+      end
+
+      # delete custom data elements from tables that may contain sensitive info
+      [
+        Hmis::Hud::CustomAssessment,
+        # other models?
```

Comment:
any model with `include ::Hmis::Hud::Concerns::FormSubmittable` could conceivable have a custom data element on it from a form. Looking at CustomAssessment, Service, Enrollment, Client, CurrentLivingSituation would be most likely to have pii

Reply:
Great, I'll add those. Thanks!